src/web/mod.rs

```diff
@@ -53,24 +53,32 @@ pub struct Application {
 impl Application {
     pub async fn build(
         settings: &Settings,
+        ldap: ldap3::Ldap,
         auth_link: String,
     ) -> Result<Application, anyhow::Error> {
         let tracing_layer = TraceLayer::new_for_http().make_span_with(|req: &Request<Body>| {
-                let method = req.method();
-                let uri = req.uri();
-                let matched_path = req.extensions().get::<MatchedPath>().map(|p| p.as_str());
+            let method = req.method();
+            let uri = req.uri();
+            let matched_path = req.extensions().get::<MatchedPath>().map(|p| p.as_str());
 
-                tracing::debug_span!("http-request", %method, %uri, matched_path, request_id = %Uuid::new_v4())
-            });
+            tracing::debug_span!("http-request", %method, %uri, matched_path, request_id = %Uuid::new_v4())
+        });
 
         let session_layer = SessionManagerLayer::new(MemoryStore::default())
             .with_secure(true)
             .with_expiry(Expiry::OnInactivity(Duration::days(7)));
 
         let registry = Registry::new(&settings.database).await?;
 
+        let users_info = UsersInfo::new(
+            ldap.clone(),
+            &settings.ldap.login,
+            &settings.ldap.password,
+            settings.ldap.users_query.clone(),
+        )
+        .await?;
         let auth_layer = AuthManagerLayerBuilder::new(
-            Backend::new(&settings.ldap, registry.clone()),
+            Backend::new(ldap, registry.clone(), users_info.clone()),
```

Comment:
if I red this correctly, the only reason to pass ldap here is to use it in auth handler. Maybe it's going to be easier to reason about if `UserInfo` struct will become general wrapper for ldap?